app/assets/javascripts/main.js

```diff
@@ -10,27 +10,40 @@ import Request from "libs/request";
 import app from "app";
 import { getWebGLReport } from "libs/webgl_stats";
 
+import React from "react";
+import ReactRouter from "react_router";
+import ReactDOM from "react-dom";
+import { Provider } from "react-redux";
+import Store from "oxalis/throttled_store";
+import { setActiveUserAction } from "oxalis/model/actions/user_actions";
+
 import "bootstrap";
 import "jasny-bootstrap";
 import "whatwg-fetch";
 import "es6-promise";
 import "libs/core_ext";
 import "backbone.marionette";
 
+import { getActiveUser } from "admin/admin_rest_api";
+
+// $FlowFixMe: CSS/LESS imports are a special WebPack feature
 import "../stylesheets/main.less";
-import Router from "./router";
 
 ErrorHandling.initialize({ throwAssertions: false, sendLocalErrors: false });
 
-app.on("start", () => {
-  app.router = new Router();
-  return Backbone.history.start({ pushState: true });
+app.on("start", async () => {
+  ReactDOM.render(
+    <Provider store={Store}>
+      <ReactRouter />
+    </Provider>,
+    document.getElementById("main-container"),
+  );
 });
 
 app.on("start", async () => {
```

Comment:
is there a reason that the `app.on("start", async () => {` is spitted up into two parts (line 34, 43)

Reply:
Yes, `app` is able to handle various callbacks at startup. Those can be completely independent of each other. Imaging if one operation takes a long time (i.e. a request to the server) the other startup callbacks can still run independently.